### ISSUE TITLE
schedutil: optionally filter job exceptions

### ISSUE THREAD

**Issue by SteVwonder** [flux-framework/flux-core#2222]
Per another thread (which I'm having trouble locating right now), @dongahn mentioned it would be helpful to have the schedutil filter out exceptions for jobs that the scheduler is unaware of (i.e., they are only in the job-manager's queue and there is no outstanding `alloc` for them).  Job's with an outstanding `alloc` request can be tracked in the schedutil_ctx added as a part of fixing #2193.

**Comment by dongahn** [comment]
Well, I was proposing not to add this at the schedutil level, though, because with this, both the scheduler and scheduleutil are tracking jobs, increasing overheads.

**Comment by garlick** [comment]
That's true, the best libschedutil could do at this point is filter out exceptions whose severity != 0.  It's just a thin layer on top of message handlers.

**Comment by garlick** [comment]
> Per another thread (which I'm having trouble locating right now),

#2220 I think?
